solr/core/src/java/org/apache/solr/servlet/RequestRateLimiter.java

```diff
@@ -91,7 +91,7 @@ boolean isEmpty() {
    * Handles an incoming request. returns a metadata object representing the metadata for the
    * acquired slot, if acquired. If a slot is not acquired, returns a null metadata object.
    */
-  public SlotReservation handleRequest() throws InterruptedException {
+  public SlotReservation handleRequest(String requestType) throws InterruptedException {
```

Comment:
I think the only upstream change we really need/want here is to make ratelimiters pluggable. We should update the `RequestRateLimiter.handleRequest()` method signature to accept `HttpServletRequest` param; then update the config parsing to accept a classname arg (analogous to other plugins) and then the logic of the plugin can be mostly separate from existing upstream code.

I have some other concerns about the generality of the priority ratelimiter implementation that would practically become non-issues if implemented as a plugin. I'm particularly eager to minimize changes to this section of the code (if possible) because the overall design of ratelimiter (at the ratelimitermanager level) is still in flux, and if we don't really need to change code here, moving to a plugin reduces the potential for future merge conflicts.

Reply:
i'm not sure about having this at plugin level given many other changes(dynamic config, initialization, etc). Also i feel this is good candidate to push upstream. Let's reconnect for it.